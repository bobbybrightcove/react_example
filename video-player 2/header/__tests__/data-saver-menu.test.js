import React from 'react';
import renderer from 'react-test-renderer';
import DataSaverMenu from '../data-saver-menu';


describe('DataSaverMenu', () => {
  const player = {};
  const onSelectDataSaverOption = jest.fn();
  const options = [
    {
      label: 'saver',
    },
    {
      label: 'sd',
    },
    {
      label: 'hd',
    },
  ];
  const selected = {
    label: 'hd',
    description: 'desc',
    bitrate: 100,
  };

  it('should render with no selection', () => {
    const component = renderer.create(
      <DataSaverMenu
        player={player}
        onSelectDataSaverOption={onSelectDataSaverOption}
        options={options}
        selected={{}}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with hd selected', () => {
    const component = renderer.create(
      <DataSaverMenu
        player={player}
        onSelectDataSaverOption={onSelectDataSaverOption}
        options={options}
        selected={selected}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
