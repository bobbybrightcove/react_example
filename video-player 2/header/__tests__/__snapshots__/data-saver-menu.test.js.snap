// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSaverMenu should render with hd selected 1`] = `
<div>
  <h4>
    Data Saver
  </h4>
  <div
    className="dataSaverOption"
    onClick={[Function]}
  >
    <div
      className="dataSaverOptionLeft"
    >
      <div
        className="dataSaverOptionName"
      >
        saver
      </div>
      <div
        className="dataSaverOptionDesc"
      />
    </div>
    <div
      className="dataSaverOptionRight"
    >
      <div
        className="checkboxWrapper"
      >
        <img
          alt={false}
          src="checkbox-unchecked.svg"
        />
      </div>
    </div>
  </div>
  <div
    className="dataSaverOption"
    onClick={[Function]}
  >
    <div
      className="dataSaverOptionLeft"
    >
      <div
        className="dataSaverOptionName"
      >
        sd
      </div>
      <div
        className="dataSaverOptionDesc"
      />
    </div>
    <div
      className="dataSaverOptionRight"
    >
      <div
        className="checkboxWrapper"
      >
        <img
          alt={false}
          src="checkbox-unchecked.svg"
        />
      </div>
    </div>
  </div>
  <div
    className="dataSaverOption"
    onClick={[Function]}
  >
    <div
      className="dataSaverOptionLeft"
    >
      <div
        className="dataSaverOptionName"
      >
        hd
      </div>
      <div
        className="dataSaverOptionDesc"
      />
    </div>
    <div
      className="dataSaverOptionRight"
    >
      <div
        className="checkboxWrapper"
      >
        <img
          alt={true}
          src="checkbox-checked.svg"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`DataSaverMenu should render with no selection 1`] = `
<div>
  <h4>
    Data Saver
  </h4>
  <div
    className="dataSaverOption"
    onClick={[Function]}
  >
    <div
      className="dataSaverOptionLeft"
    >
      <div
        className="dataSaverOptionName"
      >
        saver
      </div>
      <div
        className="dataSaverOptionDesc"
      />
    </div>
    <div
      className="dataSaverOptionRight"
    >
      <div
        className="checkboxWrapper"
      >
        <img
          alt={false}
          src="checkbox-unchecked.svg"
        />
      </div>
    </div>
  </div>
  <div
    className="dataSaverOption"
    onClick={[Function]}
  >
    <div
      className="dataSaverOptionLeft"
    >
      <div
        className="dataSaverOptionName"
      >
        sd
      </div>
      <div
        className="dataSaverOptionDesc"
      />
    </div>
    <div
      className="dataSaverOptionRight"
    >
      <div
        className="checkboxWrapper"
      >
        <img
          alt={false}
          src="checkbox-unchecked.svg"
        />
      </div>
    </div>
  </div>
  <div
    className="dataSaverOption"
    onClick={[Function]}
  >
    <div
      className="dataSaverOptionLeft"
    >
      <div
        className="dataSaverOptionName"
      >
        hd
      </div>
      <div
        className="dataSaverOptionDesc"
      />
    </div>
    <div
      className="dataSaverOptionRight"
    >
      <div
        className="checkboxWrapper"
      >
        <img
          alt={false}
          src="checkbox-unchecked.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
